File: tool/FileReader.cpp
#include <iostream>
#include <string>

class FileReader {
    private:
        std::string filename;
    public:
        std::string current_line;
        void open(std::string filename);
        void readline();
        void close();
};

void FileReader::open(std::string filename){

}

void FileReader::readline(){

}

void FileReader::close(){

}

int main(int argc, char **argv) {
    FileReader file;
    file.open("commands.txt");
    std::cout << file.current_line << std::endl;
    file.readline();
    std::cout << file.current_line << std::endl;
}

File: tool pyglet/FileReader/FIleReader.py
class FileReader:
    def __init__(self, file_path):
        self.file_path = file_path
        self.open_file()

    def open_file(self):
        self.file = open(self.file_path, 'r')

    def get_line(self):
        line = self.file.readline()
        if line == '':
            return None
        else:
            yield line
        yield from self.get_line()

    def close_file(self):
        self.file.close()
